lib/ancestry/materialized_path.rb

```diff
@@ -75,7 +75,9 @@ def siblings_of(object)
     end
 
     def ordered_by_ancestry(order = nil)
-      if %w(mysql mysql2 sqlite sqlite3 postgresql).include?(connection.adapter_name.downcase) && ActiveRecord::VERSION::MAJOR >= 5
+      if %w(mysql mysql2).include?(connection.adapter_name.downcase) && ActiveRecord::VERSION::MAJOR >= 5
+        reorder(arel_table[ancestry_column])
+      elsif %w(sqlite sqlite3 postgresql).include?(connection.adapter_name.downcase) && ActiveRecord::VERSION::MAJOR >= 5
         reorder(
           Arel::Nodes::Ascending.new(Arel::Nodes::NamedFunction.new('COALESCE', [arel_table[ancestry_column], Arel.sql("''")])),
```

Comment:
Instead of `COALESCE` you can order as per the mysql version but add `NULLS FIRST` at the end of the ordering statement. You may need to use arel to build the ordering statement so that it contains the table name to make this work with more complex queries that AR can generate sometimes.

Reply:
For all versions of postgres (again no version check), I believe you can do the same thing:

```suggestion
arel_table[ancestry_column].nulls_first
```
alternatives:
```
arel_table[ancestry_column].asc.nulls_first
Arel::Nodes::Ascending.new(arel_table[ancestry_column]).nulls_first
```

Reply:
I like `arel_table[ancestry_column].asc.nulls_first` but `asc` is probably not necessary given it's the default.

I didn't realise one could pass `nulls_first` with arel like that!